nginx/reachability_healthcheck

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+set -Eeuo pipefail
+
+# This healthcheck tests all the web services reachability from the docker
+# network (`http://<service>`) and for external server access
+# (`https://<service>.$KITSPACE_DOMAIN`) in production and staging envrionments.
+# If one of the services is unreachable it will reload nginx. If reloading
+# nginx fails it exits marks the container as unhealthy.
+
+healthcheck() {
+    # Make sure the hitting domain name returns kitspace.org and .gitea subdomain returns gitea.kitspace.org
+    test_reachability "http://frontend:3000" "Kitspace" || reload_nginx "frontend"
+    test_reachability "http://gitea:3000" "Gitea"  || reload_nginx "gitea"
+    test_reachability "http://meilisearch:7700" "Meilisearch" || reload_nginx "meilisearch"
+
+    # Run the prevoious two checks for the domain name if it's not kitspace.test
+    if [ "$KITSPACE_DOMAIN" != "kitspace.test" ]; then
+        test_reachability "https://$KITSPACE_DOMAIN" "Kitspace" || reload_nginx "The website"
+        test_reachability "https://gitea.$KITSPACE_DOMAIN" "Gitea" || reload_nginx "gitea external"
+        test_reachability "https://meilisearch.$KITSPACE_DOMAIN" "Meilisearch" || reload_nginx "meilisearch external"
+    fi
+}
+
+reload_nginx() {
+    echo "$1 is unreachable!"
+    echo "Reloading nginx!"
+    # We try to reload nginx first, if this fails we mark the container as unhealthy
+    nginx -s reload || report_healthcheck_failure "$1"
+}
```

Comment:
So this won't report anything if reloading solves it?

Reply:
Yes!

Reply:
Would be nice if it was still recorded.

Reply:
In daf9fbfa, I made it report the failure if a service is unreachable, then it tries reloading Nginx if reloading fails it reports this too.
Also, I removed the docker network checks because I verified it isn't an issue with the docker network by getting a shell inside the Nginx container and `curl`ing the container: the result was OK but the external domains were mixed.